zerver/lib/actions.py

```diff
@@ -5043,6 +5043,9 @@ def do_revoke_multi_use_invite(multiuse_invite: MultiuseInvite) -> None:
     notify_invites_changed(multiuse_invite.referred_by)
 
 def do_resend_user_invite_email(prereg_user: PreregistrationUser) -> int:
+    assert prereg_user.realm is not None
+    assert prereg_user.referred_by is not None
+
```

Comment:
I also added a `if prereg_user.referred_by is None or ` check in the caller, because these revealed a corner case where we'd 500 rather than giving a 400 with certain invalid input.

Reply:
Better to test for `None` in the gathered data than filtering, presumably?